Sources/Identity/CustomerInfoManager.swift

```diff
@@ -89,7 +90,7 @@ class CustomerInfoManager {
             }
 
             if let completion = completion {
-                self.operationDispatcher.dispatchOnMainActor {
+                self.operationDispatcher.dispatchAsyncOnMainThread {
```

Comment:
note: in this file I'm using `dispatchAsyncOnMainThread` and not `dispatchOnMainThread`. Not sure which one is best, haven't had time to dig in. But there is comment in line 307 where we're using `dispatchAsyncOnMainThread` saying that it **must** be async to prevent deadlocks, so since I didn't have time to dig in deeper I just went with that for the whole file